### PR TITLE
Generate React component for each icon

### PR DESCRIPTION
**Pull request by sandhose**
This is on top of #89 

This generates a proper React component for each icon.
This means that downstream packages can consume them directly without having to setup svgr locally.

The reason we want to do this is:

 1. this means icons can be deduplicated across packages
 2. we generate nice typescript types for each icon
 3. we have a nice index which exports all the icons
 4. less setup in downstream packages

### CODE REVIEW

File: src/@types/index.d.ts [hunk 1]
```diff
@@ -17,3 +17,11 @@ limitations under the License.
...
 
 export type Platform = "web" | "android" | "ios";
+
+declare module "prettier" {
+  export interface Options {}
+}
+
+declare module "svgo" {
+  export interface Config {}
+}
```
Comment (robintown) [inline src/@types/index.d.ts:27]: This could do with a comment like that in global.d.ts

**Review by robintown — APPROVED** [review]
Tested and working in EW. I don't think we have anyone depending on the old assets/web/icons/*.svg files that were deleted, but let's still count this as a breaking change.

File: src/@types/index.d.ts (hunk 1 above)
Comment (sandhose) [reply src/@types/index.d.ts:27]: Han, that's a leftover, I first tried there before moving things to global.d.ts

Comment (sandhose) [reply src/@types/index.d.ts]: Rebased and removed that

**Pull request merged**